server/base/src/main/java/org/apache/accumulo/server/conf/codec/GzipPropEncoding.java

```diff
@@ -0,0 +1,237 @@
+
+      // read the property map keys, values
+      Map<String,String> aMap;
+      if (opts.isCompressed()) {
+        aMap = readCompressedMap(bis);
+      } else {
+        aMap = readMap(dis);
+      }
+
+      return new VersionedPropertiesImpl(dataVersion, timestamp, aMap);
+
+    } catch (IOException ex) {
+      throw new UncheckedIOException("Encountered error deserializing properties", ex);
+    }
+  }
+
+  /**
+   * Read and uncompress an input stream compressed with GZip. The input stream is not closed by
+   * this method.
+   *
+   * @param is
+   *          an input stream
+   * @return a map with the property k.v pairs
+   * @throws IOException
+   *           if an exception occurs reading from the stream
+   */
+  private Map<String,String> readCompressedMap(final InputStream is) throws IOException {
+
+    try (GZIPInputStream gzipIn = new GZIPInputStream(is);
+        DataInputStream dis = new DataInputStream(gzipIn)) {
+      return readMap(dis);
+    }
+  }
+
+  /**
+   * Read the property map from a data input stream
+   *
+   * @param dis
+   *          a data input stream
+   * @return the property map
+   * @throws IOException
+   *           if an exception occurs reading from the stream.
+   */
+  private Map<String,String> readMap(DataInputStream dis) throws IOException {
+
+    Map<String,String> aMap = new HashMap<>();
+    int items = dis.readInt();
+
+    for (int i = 0; i < items; i++) {
+      Map.Entry<String,String> e = readKV(dis);
+      aMap.put(e.getKey(), e.getValue());
+    }
+    return aMap;
+  }
+
+  /**
+   * Write the property map to the output stream, compressing the output using GZip compression. The
+   * underlying stream will be closed when this method completes.
+   *
+   * @param os
+   *          an output stream
+   * @param aMap
+   *          the property map of k, v string pairs.
+   * @throws IOException
+   *           if an exception occurs.
+   */
+  private void writeMapCompressed(final OutputStream os, final Map<String,String> aMap)
+      throws IOException {
+
+    try (GZIPOutputStream gzipOut = new GZIPOutputStream(os);
+        DataOutputStream dos = new DataOutputStream(gzipOut)) {
+
+      writeMap(dos, aMap);
+
+      gzipOut.flush();
+      gzipOut.finish();
+
+    } catch (IOException ex) {
+      throw new IOException("Encountered error compressing properties", ex);
+    }
+  }
+
+  /**
+   * Write the property map to the data output stream. The underlying stream is not closed by this
+   * method.
+   *
+   * @param dos
+   *          a data output stream
+   * @param aMap
+   *          the property map of k, v string pairs.
+   * @throws IOException
+   *           if an exception occurs.
+   */
+  private void writeMap(final DataOutputStream dos, final Map<String,String> aMap)
+      throws IOException {
+
+    dos.writeInt(aMap.size());
+
+    aMap.forEach((k, v) -> writeKV(k, v, dos));
+
+    dos.flush();
```

Comment:
small nit. I'd clean up some of these blank lines. The more the code is spread out like that, the harder it is to fit on one screen, especially when using large fonts, making it harder to read (for me, at least)